### PR TITLE
Audit - Add Secrets & IaC to sarif format

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/utils/resultwriter.go [hunk 1]
```diff
@@ -202,27 +272,26 @@ func getCves(cvesRow []formats.CveRow, issueId string) string {
...
+func convertViolationsToSarif(jsonTable *formats.SimpleJsonResults, run *sarif.Run, markdownOutput bool) error {
 	for _, violation := range jsonTable.SecurityViolations {
-		sarifProperties, err := getSarifProperties(violation, simplifiedOutput)
+		properties, err := getViolatedDepsSarifProps(violation, markdownOutput)
 		if err != nil {
 			return err
 		}
-		err = addScanResultsToSarifRun(run, sarifProperties.Severity, violation.IssueId, sarifProperties.Headline, sarifProperties.Description, violation.Technology)
+		err = addPropertiesToSarifRun(run, &properties)
 		if err != nil {
```
Comment (eyalbe4) [inline xray/utils/resultwriter.go:286]: Let's reduce to one line.

File: xray/utils/resultwriter.go [hunk 2]
```diff
@@ -231,34 +300,36 @@ func convertViolations(jsonTable formats.SimpleJsonResults, run *sarif.Run, simp
...
+func convertVulnerabilitiesToSarif(jsonTable *formats.SimpleJsonResults, run *sarif.Run, simplifiedOutput bool) error {
 	for _, vulnerability := range jsonTable.Vulnerabilities {
-		sarifProperties, err := getSarifProperties(vulnerability, simplifiedOutput)
+		properties, err := getViolatedDepsSarifProps(vulnerability, simplifiedOutput)
 		if err != nil {
 			return err
 		}
-		err = addScanResultsToSarifRun(run, sarifProperties.Severity, vulnerability.IssueId, sarifProperties.Headline, sarifProperties.Description, vulnerability.Technology)
+		err = addPropertiesToSarifRun(run, &properties)
 		if err != nil {
```
Comment (eyalbe4) [inline xray/utils/resultwriter.go:333]: Let's reduce to one line.

File: xray/utils/resultwriter.go [hunk 3]
```diff
@@ -270,51 +341,78 @@ func convertVulnerabilities(jsonTable formats.SimpleJsonResults, run *sarif.Run,
 func getDirectDependenciesFormatted(directDependencies []formats.ComponentRow) string {
 	var formattedDirectDependencies strings.Builder
 	for _, dependency := range directDependencies {
-		formattedDirectDependencies.WriteString(fmt.Sprintf("`%s:%s`, ", dependency.Name, dependency.Version))
+		formattedDirectDependencies.WriteString(fmt.Sprintf("`%s %s`<br/>", dependency.Name, dependency.Version))
```
Comment (eyalbe4) [inline xray/utils/resultwriter.go:344]: Let's not ignore the error returned by this function.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: xray/utils/resultwriter.go (hunk 1 above)
Comment (omerzi) [reply xray/utils/resultwriter.go:286]: done

File: xray/utils/resultwriter.go (hunk 2 above)
Comment (omerzi) [reply xray/utils/resultwriter.go:333]: done

File: xray/utils/resultwriter.go (hunk 3 above)
Comment (omerzi) [reply xray/utils/resultwriter.go:344]: done

**Pull request merged**